enclave/crypto/sha.c

```diff
@@ -121,7 +121,7 @@ oe_result_t oe_sha256_restore(
 
     uint64_t NB = ((((uint64_t)num_hashed[1]) << 32) + num_hashed[0]) / 8;
     impl->ctx.total[0] = NB & 0xFFFFFFFF;
-    impl->ctx.total[1] = (NB >> 32) & 0xFFFFFFFF;
+    impl->ctx.total[1] = ((uint32_t)(NB >> 32)) & 0xFFFFFFFF;
```

Comment:
For my understanding, why is this necessary, and also, why is it not necessary for the line above this? In either case, if we're cast truncating to 32-bits, it seems the extra bit masking operation is unnecessary.

Reply:
It's necessary because GCC complains (clang doesn't). I think the issue is the automatic promotion of `(NB >> 32)` to a signed type, but I didn't investigate it any further.